### PR TITLE
Edit ws description without a modal

### PR DESCRIPTION
**Pull request by zarsky-broad**
Turns out the markdown wasn't being sanitized before, now it is, too!

See https://github.com/sparksuite/simplemde-markdown-editor/issues/243#issuecomment-313448049 for explanation.

### CODE REVIEW

File: src/pages/workspaces/workspace/Dashboard.js [hunk 1]
```diff
@@ -136,24 +88,57 @@ class WorkspaceDashboardContent extends Component {
...
+    } finally {
+      this.setState({ saving: false, editingWorkspace: false })
+    }
+  }
+
   render() {
-    const { workspace, refreshWorkspace, workspace: { accessLevel, workspace: { createdDate, lastModified, bucketName, attributes: { description } } } } = this.props
-    const { submissionsCount, storageCostEstimate, editingWorkspace } = this.state
+    const { workspace: { accessLevel, workspace: { createdDate, lastModified, bucketName, attributes: { description } } } } = this.props
+    const { submissionsCount, storageCostEstimate, editingWorkspace, editDescription = '', saving } = this.state
```
Comment (MattPutnam) [inline src/pages/workspaces/workspace/Dashboard.js]: This isn't really "editing workspace" any more, right? Also, we should be able to track both whether we're editing and the edited text as a single variable.

File: src/components/common.js [hunk 2]
```diff
@@ -179,7 +179,8 @@ export const spinnerOverlay = div(
       position: 'absolute',
       display: 'flex', alignItems: 'center',
       top: 0, right: 0, bottom: 0, left: 0,
-      backgroundColor: 'rgba(0, 0, 0, 0.1)'
+      backgroundColor: 'rgba(0, 0, 0, 0.1)',
+      zIndex: 9999 // make sure it's on top of any third party components with z-indicies
```
Comment (MattPutnam) [inline src/components/common.js:183]: I can't think of a better way to do this.

File: src/pages/workspaces/workspace/Dashboard.js [hunk 3]
```diff
@@ -136,24 +88,57 @@ class WorkspaceDashboardContent extends Component {
...
+                onChange: editDescription => this.setState({ editDescription })
+              }),
+              div({ style: { display: 'flex', justifyContent: 'flex-end', margin: '1rem' } }, [
+                buttonSecondary({ onClick: () => this.setState({ editingWorkspace: false }) }, 'Cancel'),
+                buttonPrimary({ style: { marginLeft: '1rem' }, onClick: () => this.save() }, 'Save')
+              ]),
+              saving && spinnerOverlay
+            ])
+          ],
+          [description, h(ReactMarkdown, [description])],
```
Comment (MattPutnam) [inline src/pages/workspaces/workspace/Dashboard.js]: I don't like that we have two different markdown libraries. What about using SimpleMDE in both cases, but locking it in preview mode?

File: package.json [hunk 4]
```diff
@@ -10,6 +10,7 @@
     "filesize": "^3.6.1",
     "history": "^4.7.2",
     "lodash": "^4.17.11",
+    "marked": "^0.5.1",
```
Comment (zarsky-broad) [inline package.json:13]: Was already included as a sub-dependency of simple mde.

File: src/pages/workspaces/workspace/Dashboard.js [hunk 5]
```diff
@@ -16,6 +18,9 @@ import { Component } from 'src/libs/wrapped-components'
 import { wrapWorkspace } from 'src/pages/workspaces/workspace/WorkspaceContainer'
 
 
+marked.setOptions({ sanitize: true, sanitizer: DOMPurify.sanitize })
```
Comment (panentheos) [inline src/pages/workspaces/workspace/Dashboard.js]: If this is effectively a global setting, it might be nice to put this in a more central location.

**Review by panentheos — APPROVED** [review]
Looks good.

**Pull request merged**